### PR TITLE
Add Virtual Switch skeleton

### PR DESCRIPTION
**Pull request by lguohan**
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/lguohan/sonic-sairedis/pull/1%23discussion_r73914612%22%2C%20%22https%3A//github.com/lguohan/sonic-sairedis/pull/1%23discussion_r73919514%22%2C%20%22https%3A//github.com/lguohan/sonic-sairedis/pull/1%23discussion_r74285160%22%2C%20%22https%3A//github.com/lguohan/sonic-sairedis/pull/1%23issuecomment-238932725%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/lguohan/sonic-sairedis/pull/1%23issuecomment-238932725%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Ack.%20do%20you%20want%20to%20generate%20pull%20request%20against%20the%20azure%20repo%3F%22%2C%20%22created_at%22%3A%20%222016-08-10T17%3A01%3A22Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/2293967%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/lguohan%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%2064e3a96cd907dc1013e2217f4e435afc3c28a287%20vslib/src/sai_vs_generic_create.cpp%2060%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/lguohan/sonic-sairedis/pull/1%23discussion_r73914612%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22object%20id%20is%20out%20only%2C%20no%20need%20to%20find%3F%20Assume%20you%20won%27t%20find%20it.%22%2C%20%22created_at%22%3A%20%222016-08-08T17%3A13%3A42Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/2293967%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/lguohan%22%7D%7D%2C%20%7B%22body%22%3A%20%22It%20out%20param%2C%20take%20a%20look%20at%20line%20108%20and%20117%20and%20122%2C%20%5Cr%5Cnalso%20look%20at%20line%2064%2C%20this%20is%20just%20safety%20check%22%2C%20%22created_at%22%3A%20%222016-08-08T17%3A43%3A07Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/17052279%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kcudnik%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20vslib/src/sai_vs_generic_create.cpp%3AL1-214%22%7D%2C%20%22Pull%201d8c5445c28a2ff41cebc965d1c153309563b795%20lib/src/Makefile.am%206%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/lguohan/sonic-sairedis/pull/1%23discussion_r74285160%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22a%20minor%20issue%2C%20we%20probably%20want%20to%20keep%20the%20old%20path%20for%20the%20sai%3F%22%2C%20%22created_at%22%3A%20%222016-08-10T17%3A00%3A12Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/2293967%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/lguohan%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20lib/src/Makefile.am%3AL1-9%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [ ] <a href='#crh-comment-Pull 64e3a96cd907dc1013e2217f4e435afc3c28a287 vslib/src/sai_vs_generic_create.cpp 60'></a> &nbsp;<b><a href='https://github.com/lguohan/sonic-sairedis/pull/1#discussion_r73914612'>File: vslib/src/sai_vs_generic_create.cpp:L1-214</a></b>
- <a href='https://github.com/lguohan'></a> object id is out only, no need to find? Assume you won't find it.
- <a href='https://github.com/kcudnik'></a> It out param, take a look at line 108 and 117 and 122,
  also look at line 64, this is just safety check
- [ ] <a href='#crh-comment-Pull 1d8c5445c28a2ff41cebc965d1c153309563b795 lib/src/Makefile.am 6'></a> &nbsp;<b><a href='https://github.com/lguohan/sonic-sairedis/pull/1#discussion_r74285160'>File: lib/src/Makefile.am:L1-9</a></b>
- <a href='https://github.com/lguohan'></a> a minor issue, we probably want to keep the old path for the sai?
- [ ] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/lguohan/sonic-sairedis/pull/1#issuecomment-238932725'>General Comment</a></b>
- <a href='https://github.com/lguohan'></a> Ack. do you want to generate pull request against the azure repo?

<a href='https://www.codereviewhub.com/lguohan/sonic-sairedis/pull/1?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/lguohan/sonic-sairedis/pull/1?approve=1'></a>&nbsp;<a href='https://github.com/lguohan/sonic-sairedis/pull/1'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by lguohan** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/lguohan/sonic-sairedis/pull/1'>Refresh</a>.<a href='#crh-update-none'></a>

File: vslib/src/sai_vs_generic_create.cpp [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+        _In_ sai_object_type_t object_type,
+        _In_ const std::string &serialized_object_id,
+        _In_ uint32_t attr_count,
+        _In_ const sai_attribute_t *attr_list)
+{
+    std::lock_guard<std::mutex> lock(g_mutex);
+
+    SWSS_LOG_ENTER();
+
+    auto it = g_objectHash.find(serialized_object_id);
```
Comment (lguohan) [inline vslib/src/sai_vs_generic_create.cpp:60]: object id is out only, no need to find? Assume you won't find it.

Comment (kcudnik) [inline vslib/src/sai_vs_generic_create.cpp:60]: It out param, take a look at line 108 and 117 and 122, 
also look at line 64, this is just safety check

File: lib/src/Makefile.am [hunk 2]
```diff
@@ -1,7 +1,8 @@
 # Makefile.am -- Process this file with automake to produce Makefile.in
 
 AM_CPPFLAGS =
-AM_CPPFLAGS += -I/usr/include/sai -I/usr/include/swss -I$(top_srcdir)/common
+#AM_CPPFLAGS += -I/usr/include/sai -I/usr/include/swss -I$(top_srcdir)/common
+AM_CPPFLAGS += -I/home/kcudnik/Networking/SAI/inc -I/usr/include/swss -I$(top_srcdir)/common
```
Comment (lguohan) [inline lib/src/Makefile.am]: a minor issue, we probably want to keep the old path for the sai?

**Comment by lguohan** [comment]
Ack. do you want to generate pull request against the azure repo?

**Pull request closed without merging**